specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json

```diff
@@ -1526,6 +1526,51 @@
         }
       }
     },
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountName}/createEnrollmentAccount": {
+      "post": {
```

Comment:
>post [](start = 7, length = 4)

Why are we not modeling it as PUT? Please see [https://armwiki.azurewebsites.net/api_contracts/guidelines/templatedeployment.html#td001-use-put-for-resource-provisioning](https://armwiki.azurewebsites.net/api_contracts/guidelines/templatedeployment.html#td001-use-put-for-resource-provisioning)

Reply:
The path of GET api is /providers/Microsoft.Billing/billingAccounts/{billingAccountName}/enrollmentAccounts/{enrollmentAccountName}
Even though the path says {enrollmentAccountName} the value is a system generated identifier. All the other existing apis are designed in the same way. If PUT is used to create an enrollment account, then the name would become part of the URL, this would be confusing to the user as the would be trying to pass the name to do the GET but the value would be a system generated identifier. Also in PUT, if there are certain value that cannot be updated like the account owner email and authentication type, which can be set during the create (POST). If we changed the GET for this particular enrollment accounts, there are lots of other apis in billing.json which follow the same pattern of passing identifier for name, which will result in confusion.